packages/hawtio/src/plugins/connect/jolokia-service.ts

```diff
@@ -406,10 +405,13 @@ class JolokiaService implements IJolokiaService {
 
   async search(mbeanPattern: string): Promise<string[]> {
     const jolokia = await this.jolokia
-    return new Promise(resolve => {
+    return new Promise((resolve, reject) => {
       jolokia.search(
         mbeanPattern,
-        onSearchSuccess((response: string[]) => resolve(response)),
+        onSimpleSuccessAndError(
+          response => resolve(response as string[]),
+          (response: IErrorResponse) => reject(response.stacktrace || response.error),
```

Comment:
I would avoid using `reject` anywhere in the hawtio-next, as it would be interpreted as an error thrown which hawtio couldn't handle. I'd rather output `log.error(<some error message>)` and resolve `[] as string[]`.